src/main/scala/io/iohk/ethereum/mining/BlockGenerator.scala

```diff
@@ -75,6 +73,22 @@ class BlockGenerator(blockchainStorages: BlockchainStorages, blockchainConfig: B
         }
       }.getOrElse(Left(NoParent))
     }
+
+    result.right.foreach(b => cache.updateAndGet(new UnaryOperator[List[Block]] {
+      override def apply(t: List[Block]): List[Block] =
+        (b :: t).take(cacheLimit)
+    }))
```

Comment:
Say we continuously receive new transaction and submit new blocks for mining, without including the mining results. Yes we can mine them, but without keeping the order (the parent - child relation) those block cannot form a chain, so we waste resources.